src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (type == Bulldozer.class) {
+            List<?> result = new BulldozerProducerImpl().get();
+            return new BulldozerProducerImpl().get();
+        }
+
+        if (type == Excavator.class) {
+            return new ExcavatorProducerImpl().get();
+        }
+
+        if (type == Truck.class) {
+            return new TruckProducerImpl().get();
+        }
+
+        return List.of();
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        for (int i = 0; i < machines.size(); i++) {
+            machines.set(i, value);
+        }
+    }
+
+    @Override
+    public void startWorking(List<? extends Machine> machines) {
+
+    }
```

Comment:
The `startWorking` method is not implemented. You should iterate over the `machines` list and call the `doWork()` method on each machine to fulfill the method's contract.